pysmt/simplifier.py

```diff
@@ -593,7 +593,7 @@ def walk_bv_ashr(self, formula, args, **kwargs):
             return ret
         return self.manager.BVAShr(l, r)
 
-    def walk_length(self, formula, args, **kwargs):
-        return self.manager.Length(args[0])
+    def walk_str_length(self, formula, args, **kwargs):
+        return self.manager.str_length(args[0])
 
```

Comment:
In the simplest form, it should be possible to apply simplification rules on constants. 
For example, length of a constant can by computed via python. Similarly, methods such as concatenation etc.

More complex simplifications can be left as future work, and implemented as needed.